### PR TITLE
Eagle: MM Cuda Graphs with MRope

### PR DESCRIPTION
**Pull request by IzzyPutterman**
## Purpose

Enabling MRope for Eagle in https://github.com/vllm-project/vllm/pull/22872 broke torch compile, this was due to the fact that the positions now have dynamic shapes in last dim instead of dim=0.  Porting that change from other MM models results torch compile + cuda graphs running.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/v1/spec_decode/eagle.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def __init__(
         if self.uses_mrope:
             # M-RoPE need (3, max_num_tokens)
             self.mrope_positions = torch.zeros(
-                (3, self.max_num_tokens), dtype=torch.int64, device=device
+                (3, self.max_num_tokens + 1), dtype=torch.int64, device=device
```
Comment (benchislett) [inline vllm/v1/spec_decode/eagle.py:130]: Context for this change is here:
https://github.com/vllm-project/vllm/blob/4c23690f43e51eccf6ce5866ac47adcf39215e4d/vllm/v1/worker/gpu_model_runner.py#L491

**Review by benchislett — APPROVED** [review]
LGTM
